fuel-vm/src/constraints.rs

```diff
@@ -181,16 +16,28 @@ pub struct InstructionLocation {
     pub offset: u64,
 }
 
-impl<const LEN: usize> Deref for CheckedMemConstLen<LEN> {
-    type Target = CheckedMemRange;
+impl InstructionLocation {
+    /// New location from context and offset
+    pub const fn new(context: Option<ContractId>, offset: u64) -> Self {
+        Self { context, offset }
+    }
+
+    /// Context, i.e. current contract
+    pub const fn context(&self) -> Option<ContractId> {
+        self.context
+    }
 
-    fn deref(&self) -> &Self::Target {
-        &self.0
+    /// Offset from the IS register
+    pub const fn offset(&self) -> u64 {
+        self.offset
     }
 }
 
-impl<const LEN: usize> DerefMut for CheckedMemConstLen<LEN> {
-    fn deref_mut(&mut self) -> &mut Self::Target {
-        &mut self.0
+impl<S, Tx> Interpreter<S, Tx> {
+    pub(crate) fn current_location(&self) -> InstructionLocation {
```

Comment:
Previously it was under `#[cfg(feature = "profile-gas")]` flag and didn't participate in the production. But now, it will. I think we need to put fields under this flag too(or use different `InstructionLocation` for each flag)